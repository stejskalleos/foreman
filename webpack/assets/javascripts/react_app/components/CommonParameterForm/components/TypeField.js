import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from '../../../common/I18n';
import FormField from '../../common/forms/FormField';

import { COMMON_PARAM_TYPES } from '../CommonParamFormConsts';

const TypeField = ({ selectedType, onChange }) => {
  const helpContent = (
    <Fragment>
      <dt>String</dt>
      <dd>{__('Everything turns into a string.')}</dd>
      <dt>Boolean</dt>
      <dd>{__('true / false')}</dd>
      <dt>Integer</dt>
      <dd>{__('Integer numbers only, can be negative.')}</dd>
      <dt>Real</dt>
      <dd>{__('Accept any numerical input.')}</dd>
      <dt>Array</dt>
      <dd>
        {__('A valid JSON or YAML input, that must evaluate to an array.')}
        <br />
        {__('Example')}:&nbsp;<code>[&quot;some-value&quot;, 123, true]</code>
      </dd>
      <dt>Hash</dt>
      <dd>
        {__(
          'A valid JSON or YAML input, that must evaluate to an object/map/dict/hash.'
        )}
      </dd>
      <dd>
        {__('Examples')}:<br />
        <code>
          &#123;&quot;key&quot; =&#62; &quot;value&quot;&#125;
          <br />
          key: value
        </code>
      </dd>
      <dt>YAML</dt>
      <dd>
        {__('Any valid YAML input.')} {__('Example')}:&nbsp;
        <code>key: value</code>
      </dd>
      <dt>JSON</dt>
      <dd>
        {__('Any valid JSON input.')} {__('Example')}:&nbsp;
        <code>&#123;&quot;key&quot;: &quot;value&quot;&#125;</code>
      </dd>
    </Fragment>
  );

  return (
    <FormField label={__('Parameter Type')} labelHelp={helpContent}>
      <select
        id="common_parameter_parameter_type"
        name="common_parameter[parameter_type]"
        value={selectedType}
        onChange={e => onChange(e.target.value)}
        className="form-control without_select2"
      >
        {COMMON_PARAM_TYPES.map(item => (
          <option value={item.value} key={item.value}>
            {item.label}
          </option>
        ))}
      </select>
    </FormField>
  );
};

TypeField.propTypes = {
  selectedType: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

TypeField.defaultProps = {
  selectedType: '',
};

export default TypeField;
