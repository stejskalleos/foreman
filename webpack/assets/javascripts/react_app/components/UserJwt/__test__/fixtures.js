import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
