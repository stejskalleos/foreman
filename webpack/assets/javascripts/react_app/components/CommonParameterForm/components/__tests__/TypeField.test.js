import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { typeFieldFixtures } from './fixtures';
import TypeField from '../TypeField';

describe('TypeField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TypeField, typeFieldFixtures));
});
