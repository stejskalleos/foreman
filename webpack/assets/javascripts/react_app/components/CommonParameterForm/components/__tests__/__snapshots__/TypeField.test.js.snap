// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeField rendering default 1`] = `
<FormField
  className=""
  disabled={false}
  error={null}
  helpInline={null}
  id={null}
  inputProps={null}
  inputSizeClass="col-md-4"
  label="Parameter Type"
  labelHelp={
    <React.Fragment>
      <dt>
        String
      </dt>
      <dd>
        Everything turns into a string.
      </dd>
      <dt>
        Boolean
      </dt>
      <dd>
        true / false
      </dd>
      <dt>
        Integer
      </dt>
      <dd>
        Integer numbers only, can be negative.
      </dd>
      <dt>
        Real
      </dt>
      <dd>
        Accept any numerical input.
      </dd>
      <dt>
        Array
      </dt>
      <dd>
        A valid JSON or YAML input, that must evaluate to an array.
        <br />
        Example
        : 
        <code>
          ["some-value", 123, true]
        </code>
      </dd>
      <dt>
        Hash
      </dt>
      <dd>
        A valid JSON or YAML input, that must evaluate to an object/map/dict/hash.
      </dd>
      <dd>
        Examples
        :
        <br />
        <code>
          {"key" =&gt; "value"}
          <br />
          key: value
        </code>
      </dd>
      <dt>
        YAML
      </dt>
      <dd>
        Any valid YAML input.
         
        Example
        : 
        <code>
          key: value
        </code>
      </dd>
      <dt>
        JSON
      </dt>
      <dd>
        Any valid JSON input.
         
        Example
        : 
        <code>
          {"key": "value"}
        </code>
      </dd>
    </React.Fragment>
  }
  labelSizeClass="col-md-2"
  onChange={[Function]}
  required={false}
  type="text"
>
  <select
    className="form-control without_select2"
    id="common_parameter_parameter_type"
    name="common_parameter[parameter_type]"
    onChange={[Function]}
    value="string"
  >
    <option
      key="string"
      value="string"
    >
      String
    </option>
    <option
      key="boolean"
      value="boolean"
    >
      Boolean
    </option>
    <option
      key="integer"
      value="integer"
    >
      Integer
    </option>
    <option
      key="real"
      value="real"
    >
      Real
    </option>
    <option
      key="array"
      value="array"
    >
      Array
    </option>
    <option
      key="hash"
      value="hash"
    >
      Hash
    </option>
    <option
      key="yaml"
      value="yaml"
    >
      Yaml
    </option>
    <option
      key="json"
      value="json"
    >
      JSON
    </option>
  </select>
</FormField>
`;
